lib/eventsource/index.js

```diff
@@ -0,0 +1,448 @@
+      this.#readyState = OPEN
+      this.dispatchEvent(new Event('open', {}))
+
+      const eventSourceStream = new EventSourceStream({
+        eventSourceState: this.#settings,
+        push: (eventPayload) => {
+          this.dispatchEvent(new MessageEvent(
+            eventPayload.type,
+            eventPayload.payload
+          ))
+        }
+      })
+
+      pipeline(response.body.stream,
+        eventSourceStream,
+        (error) => {
+          if (
+            error &&
+            error.aborted === false
+          ) {
+            this.close()
+            this.dispatchEvent(new ErrorEvent('error', { error }))
+          }
+        })
+    }
+
+    fetchParam.timingInfo = createOpaqueTimingInfo({})
+
+    fetchParam.controller = new Fetch(getGlobalDispatcher())
+
+    this.#controller = fetchParam.controller
+
+    await mainFetch(fetchParam)
+  }
+
+  /**
+   * @see https://html.spec.whatwg.org/multipage/server-sent-events.html#sse-processing-model
+   * @returns {Promise<void>}
+   */
+  async #reconnect () {
+    // When a user agent is to reestablish the connection, the user agent must
+    // run the following steps. These steps are run in parallel, not as part of
+    // a task. (The tasks that it queues, of course, are run like normal tasks
+    // and not themselves in parallel.)
+
+    // 1. Queue a task to run the following steps:
+
+    //   1. If the readyState attribute is set to CLOSED, abort the task.
+    if (this.#readyState === CLOSED) return
+
+    //   2. Set the readyState attribute to CONNECTING.
+    this.#readyState = CONNECTING
+
+    //   3. Fire an event named error at the EventSource object.
+    this.dispatchEvent(new ErrorEvent('error', { message: 'Reconnecting' }))
+
+    // 2. Wait a delay equal to the reconnection time of the event source.
+    await setTimeout(this.#settings.reconnectionTime)
+
+    // 5. Queue a task to run the following steps:
+
+    //   1. If the EventSource object's readyState attribute is not set to
+    //      CONNECTING, then return.
+    if (this.#readyState !== CONNECTING) return
+
+    //   2. Let request be the EventSource object's request.
+    //   3. If the EventSource object's last event ID string is not the empty
+    //      string, then:
+    //      1. Let lastEventIDValue be the EventSource object's last event ID
+    //         string, encoded as UTF-8.
+    //      2. Set (`Last-Event-ID`, lastEventIDValue) in request's header
+    //         list.
+    if (this.#settings.lastEventId !== '') {
+      this.#request.headers.set('last-event-id', this.#settings.lastEventId, true)
+    }
+
+    //   4. Fetch request and process the response obtained in this fashion, if any, as described earlier in this section.
+    this.#connect()
+  }
+
+  /**
+   * Closes the connection, if any, and sets the readyState attribute to
+   * CLOSED.
+   */
+  close () {
+    webidl.brandCheck(this, EventSource)
```

Comment:
no need, the private properties act as a brand check for us

Reply:
This is not a private property?! Are you sure?